Tests/SwiftASN1Tests/UTCTimeTests.swift

```diff
@@ -0,0 +1,59 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftASN1 open source project
+//
+// Copyright (c) 2023 Apple Inc. and the SwiftASN1 project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftASN1 project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+import XCTest
+
+@testable import SwiftASN1
+
+final class UTCTimeTests: XCTestCase {
+    func testComparisons() throws {
+        let original = try UTCTime(year: 2020, month: 03, day: 03, hours: 03, minutes: 03, seconds: 03)
+
+        func modify(_ field: WritableKeyPath<UTCTime, Int>, of time: UTCTime, by modifier: Int) -> UTCTime {
+            var copy = time
+            copy[keyPath: field] += modifier
+            return copy
+        }
+
+        let integerTransformable: [WritableKeyPath<UTCTime, Int>] = [
+            \.year, \.month, \.day, \.hours, \.minutes, \.seconds
+        ]
+
+        var transformationsAndResults: [(UTCTime, lt: Bool, eq: Bool)] = []
```

Comment:
Nit: let's combine `lt` and `eq` into an enum e.g. 
```swift
enum ExpectedComparisionResult {
    case equal
    case smaller
    case greater
}
```
```suggestion
        var transformationsAndResults: [(UTCTime, expectedResult: ExpectedComparisionResult)] = []
```

Reply:
Good idea, I've made that change.